src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,130 @@
+import { useState, useEffect, useMemo } from 'react';
+import { useMatch, useSearchParams } from 'react-router-dom';
+import { Person } from '../types/Person';
+import { getPeople } from '../api';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  const match = useMatch('/people/:personId');
+  const selectedPersonId = match?.params.personId || '';
+  const [searchParams] = useSearchParams();
+  const currentGender = searchParams.get('sex');
+  const pickedCenturies = searchParams.getAll('centuries');
+  const sortName = searchParams.get('sort');
+  const order = searchParams.get('order');
+  const query = searchParams.get('query');
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople().then(result => {
+      setPeople(result);
+    })
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const visiblePeople = useMemo(() => {
+    const filteredArray = (currentGender)
+      ? people.filter(person => (person.sex === currentGender))
+      : [...people];
+    const filteredByCentury = (pickedCenturies.length > 0)
+      ? filteredArray.filter(person => (
+        pickedCenturies.includes(Math.ceil(person.born / 100).toString())))
+      : filteredArray;
+
+    if (query !== '' && query !== null) {
+      const normalizedQuery = query.toLowerCase();
+
+      return filteredByCentury.filter(person => (
+        person.name.toLowerCase().includes(normalizedQuery)
+          || person.motherName?.toLowerCase().includes(normalizedQuery)
+          || person.fatherName?.toLowerCase().includes(normalizedQuery)));
+    }
+
+    return filteredByCentury;
+  }, [currentGender, pickedCenturies, people, query]);
+
+  const sortedPeople = useMemo(() => {
+    if (sortName) {
+      return [...visiblePeople].sort((a, b) => {
+        const curr = a[sortName as keyof typeof a];
+        const next = b[sortName as keyof typeof b];
```

Comment:
Doesn't it work without it? Just curious 🤔 
```suggestion
        const curr = a[sortName];
        const next = b[sortName];
```

Reply:
it worked, but i have error there and can't deploy